OpenSim/Common/TRCFileAdapter.cpp

```diff
@@ -162,12 +192,16 @@ TRCFileAdapter::extendRead(const std::string& fileName) const {
 
         // Columns 2 till the end are data.
         TimeSeriesTableVec3::RowVector 
-            row_vector{static_cast<int>(num_markers_expected)};
+            row_vector{static_cast<int>(num_markers_expected), 
+                       SimTK::Vec3(SimTK::NaN)};
         int ind{0};
-        for(std::size_t c = 2; c < column_labels.size() * 3 + 2; c += 3)
-            row_vector[ind++] = SimTK::Vec3{std::stod(row.at(c)),
-                                            std::stod(row.at(c+1)),
-                                            std::stod(row.at(c+2))};
+        for (std::size_t c = 2; c < column_labels.size() * 3 + 2; c += 3) {
+            if (!row.at(c).empty())
```

Comment:
Can you use `{}` for this `if`? That'll help make it clearer that you want `++ind` outside the `if`.